### ISSUE TITLE
OneOf used incorrectly in OpenAPI definition

### ISSUE THREAD

**Issue by rwb27** [labthings/python-labthings#248]
**Describe the bug**
Actions are described by a schema, in the OpenAPI/Swagger export, that includes:
```
                  status:
                    OneOf:
                    - pending
                    - running
                    - completed
                    - cancelled
                    - error
                    type: string
```
This is not what `OneOf` means in OpenAPI, and it's not valid.  This means the OpenAPI description is invalid, and stops a bunch of tools working with it.

**To Reproduce**
Export a YAML or JSON description of the API and try to validate it.

**Expected behaviour**
The `status` field should be serialised in the JSON schema as
```
status:
  type: string
  enum:
    - pending
    - running
    - completed
    - cancelled
    - error
```

**Additional context**
I think this comes from the line that defines the field:
```
    _status = fields.String(
        data_key="status",
       OneOf =["pending", "running", "completed", "cancelled", "error"],
    )
```
I suspect this was intended to set a `OneOf` validator on the field.  Instead what it seems to be doing is attaching additional metadata to the field, that is passed directly into the OpenAPI spec.

**Comment by rwb27** [comment]
I will add this to a documentation-related PR.

**Comment by jtc42** [comment]
> I will add this to a documentation-related PR.

Thanks!

**Comment by rwb27** [comment]
Closed by #249
